BootloaderCorePkg/Include/Library/MpInitLib.h

```diff
@@ -46,6 +46,29 @@ typedef struct {
 typedef UINT32 (*CPU_TASK_PROC)          (UINT32 Arg);
 typedef VOID   (*PLATFORM_CPU_INIT_HOOK) (UINT32 CpuIndex);
 
+#define SMM_BASE_GAP        0x1000
+#define SMM_BASE_MIN_SIZE   0x10000
+
+#define SMM_BASE_INFO_SIG   SIGNATURE_32('S', 'C', 'O', 'M')
+
+#pragma pack(1)
+
+typedef struct {
+  UINT32  ApicId;
+  UINT32  SmmBase;
+} CPU_SMMBASE;
+
+typedef struct {
+  UINT32        Signature;
+  UINT8         CpuCnt;
+  UINT8         Unused;
+  UINT16        Rsvd;
+  CPU_SMMBASE   SmmBase[];
+} SMMBASE_INFO;
+
+#pragma pack()
+
+

```

Comment:
Given added a new library S3SaveRestoreLib, why S3SaveRestoreLib.h is missing?
Prefer moving these definition to S3SaveRestoreLib.h

Reply:
In the previous discussions, it was decided that this was CPU related info and a better place was in MpInitLib.h

Reply:
Since now we have S3SaveRestoreLib, does it make more sense to move to S3SaveRestoreLib?

Reply:
In that case, MpInitLib has to include S3SaveRestoreLib header. Ideally it does not need to know S3 related information. So, decided to separate S3 related structures from including in MP related files. Please let me know if you have still have a concern.